### PR TITLE
Add pickler for BuildDependencies

### PR DESCRIPTION
**Pull request by dancingrobot84**

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @dancingrobot84,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

**Comment by typesafehub-validator** [comment]
Hi @dancingrobot84,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://typesafe.com/contribute/cla](http://typesafe.com/contribute/cla)

**Comment by jsuereth** [comment]
The change looks good (minus the build failure).  What i'm debating right now is whether or not we should just return this info in the MinimalBuildStructure/MinimalProjectStructure, as I think it's valuable there...

File: commons/protocol/src/main/scala/sbt/protocol/Keys.scala [hunk 1]
```diff
@@ -98,3 +98,19 @@ object MinimalBuildStructure {
   implicit val unpickler: Unpickler[MinimalBuildStructure] = genUnpickler[MinimalBuildStructure]
   implicit val pickler: Pickler[MinimalBuildStructure] = genPickler[MinimalBuildStructure]
 }
+
+final case class BuildDependencies(
+  classpath: Map[ProjectReference, Seq[ClasspathDep]],
+  aggregate: Map[ProjectReference, Seq[ProjectReference]])
+object BuildDependencies {
+  implicit val unpickler: Unpickler[BuildDependencies] = genUnpickler[BuildDependencies]
+  implicit val pickler: Pickler[BuildDependencies] = genPickler[BuildDependencies]
```
Comment (havocp) [inline commons/protocol/src/main/scala/sbt/protocol/Keys.scala]: These two vals can be one val `implicit val picklerUnpickler = PicklerUnpickler.generate[BuildDependencies]` now; there are a bunch of examples like the above leftover from before we had PicklerUnpickler.

**Comment by dpratt** [comment]
Adding this to Minimal(Project|Build)Structure would be incredibly useful for me - I've been working on an IntelliJ sbt-rc plugin and having this available immediately at project loading time would make the code paths that construct the relevant IntelliJ model bits quite simpler.

**Comment by jsuereth** [comment]
Hey @dpratt   / @dancingrobot84  Sorry for the delay.     I'd like to adapt this PR so that the project deps show up in minimal build structure.  So, we can leave this as reference but I plan to eventually close in favor of that approach.

@dancingrobot84 didn't chime in, but will that solve the issue you need us to solve?

**Comment by dancingrobot84** [comment]
@jsuereth Yeah, it will solve.

**Comment by jsuereth** [comment]
Closing in favor of #289

**Pull request closed without merging**